chaincode/fabcar/java/src/test/java/org/hyperledger/fabric/samples/fabcar/CarQueryResultTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.fabric.samples.fabcar;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+public final class CarQueryResultTest {
+
+    @Nested
+    class Equality {
+
+        @Test
+        public void isReflexive() {
+            CarQueryResult cqr = new CarQueryResult("CAR1", new Car("Toyota", "Prius", "blue", "Tomoko"));
+
+            assertThat(cqr).isEqualTo(cqr);
+        }
+
+        @Test
+        public void isSymmetric() {
+            Car car = new Car("Toyota", "Prius", "blue", "Tomoko");
+            CarQueryResult cqrA = new CarQueryResult("CAR1", car);
+            CarQueryResult cqrB = new CarQueryResult("CAR1", car);
+
+            assertThat(cqrA).isEqualTo(cqrB);
+            assertThat(cqrB).isEqualTo(cqrA);
+        }
+
+        @Test
+        public void isTransitive() {
+            Car car = new Car("Toyota", "Prius", "blue", "Tomoko");
+            CarQueryResult cqrA = new CarQueryResult("CAR1", car);
+            CarQueryResult cqrB = new CarQueryResult("CAR1", car);
+            CarQueryResult cqrC = new CarQueryResult("CAR1", car);
+
+            assertThat(cqrA).isEqualTo(cqrB);
+            assertThat(cqrB).isEqualTo(cqrC);
+            assertThat(cqrA).isEqualTo(cqrC);
+        }
+
+        @Test
+        public void handlesKeyInequality() {
+            CarQueryResult cqrA = new CarQueryResult("CAR1", new Car("Toyota", "Prius", "blue", "Tomoko"));
+            CarQueryResult cqrB = new CarQueryResult("CAR2", new Car("Toyota", "Prius", "blue", "Tomoko"));
+
+            assertThat(cqrA).isNotEqualTo(cqrB);
+        }
+
+        @Test
+        public void handlesRecordInequality() {
+            CarQueryResult cqrA = new CarQueryResult("CAR1", new Car("Toyota", "Prius", "blue", "Tomoko"));
+            CarQueryResult cqrB = new CarQueryResult("CAR1", new Car("Ford", "Mustang", "red", "Brad"));
+
+            assertThat(cqrA).isNotEqualTo(cqrB);
+        }
+
+        @Test
+        public void handlesKeyRecordInequality() {
+            CarQueryResult cqrA = new CarQueryResult("CAR1", new Car("Toyota", "Prius", "blue", "Tomoko"));
+            CarQueryResult cqrB = new CarQueryResult("CAR2", new Car("Ford", "Mustang", "red", "Brad"));
+
+            assertThat(cqrA).isNotEqualTo(cqrB);
+        }
+
+        @Test
+        public void handlesOtherObjects() {
+            CarQueryResult cqrA = new CarQueryResult("CAR1", new Car("Toyota", "Prius", "blue", "Tomoko"));
+            String cqrB = "not a car";
+
+            assertThat(cqrA).isNotEqualTo(cqrB);
+        }
+
+        @Test
+        public void handlesNull() {
+            CarQueryResult cqr = new CarQueryResult("CAR1", new Car("Toyota", "Prius", "blue", "Tomoko"));
+
+            assertThat(cqr).isNotEqualTo(null);
+        }
+    }
+
+    @Test
+    public void toStringIdentifiesCarQueryResult() {
```

Comment:
is a test on toString() really necessary?  specially asserting on the hash code

toString() is meant for display/logging. If you agree, then let us just remove it from CarTest as well.

Reply:
I would disagree; toString() is part of the external API, so it's proper to have tests for it.  

There's nothing worth than debugging a log from a production system only to find you don't have the correct information.

Reply:
It was also required for the coverage check to pass- it's a small sample so it seems like 100% coverage is reasonable to aspire to